packages/next/next-server/server/next-server.ts

```diff
@@ -978,6 +978,9 @@ export default class Server {
       ? (req as any)._nextRewroteUrl
       : `${parseUrl(req.url || '').pathname!}`
 
+    // remove trailing slash
+    urlPathname = urlPathname.replace(/(?!^)\/$/, '')
```

Comment:
Is this change out of scope re: your future PR fixing trailing slash behaviors? Or is it easier to just do this now?

I don't see adequate test coverage for this behavior.

Reply:
This is needed by both PRs for different purposes, and covered by two different tests, one in each PR.
tested here: https://github.com/vercel/next.js/pull/13559/files#diff-6a1ab8ce5323e3179038754e716d22baR98
and here: https://github.com/vercel/next.js/pull/13333/files#diff-4f5701207fbffaffe78cae6aa30571a4R56

Reply:
Please extend this test:
https://github.com/vercel/next.js/pull/13559/files#diff-6a1ab8ce5323e3179038754e716d22baR98

To include a subpath as well.

Reply:
like this test? https://github.com/vercel/next.js/pull/13559/files#diff-6a1ab8ce5323e3179038754e716d22baR104